clion/src/main/kotlin/org/rust/clion/profiler/perf/RsPerfConfigurationExtension.kt

```diff
@@ -45,6 +46,7 @@ class RsPerfConfigurationExtension : CargoCommandConfigurationExtension() {
         val project = configuration.project
         val settings = CPPProfilerSettings.instance.state
         val perfPath = settings.executablePath.orEmpty()
+        validatePerfSettings(project)?.let { throw it }
```

Comment:
A quite weird construction. Maybe it's better just to throw exception in `validatePerfSettings` if needed and not to return exception object at all?

Reply:
`validatePerfSettings` uses three API functions that return exception objects. Should I use something like that?

```Kotlin
private fun validatePerfSettings(project: Project) {
    val state = CPPProfilerSettings.instance.state
    val settingsClass = profilerConfigurable::class.java

    throw validateLocalPath(state.executablePath.orEmpty(), "Perf executable", project, settingsClass)
        ?: validateFrequency(state.samplingFrequency, project, settingsClass)
        ?: validateOutputDirectory(state.outputDirectory.orEmpty(), project, settingsClass)
        ?: return
}
```

More straightforward Java-style construction looks too bulky to me:
```Java
private fun validatePerfSettings(project: Project) {
    val state = CPPProfilerSettings.instance.state
    val settingsClass = profilerConfigurable::class.java
    
    val localPathValidationException = validateLocalPath(state.executablePath.orEmpty(), "Perf executable", project, settingsClass)
    if (localPathValidationException != null) {
        throw localPathValidationException
    }
    val frequencyValidationException = validateFrequency(state.samplingFrequency, project, settingsClass)
    if (frequencyValidationException != null) {
        throw frequencyValidationException
    }
    val outputDirValidationException = validateOutputDirectory(state.outputDirectory.orEmpty(), project, settingsClass)
    if (outputDirValidationException != null) {
        throw outputDirValidationException
    }
}
```